rust/src/tests/chown.rs

```diff
@@ -0,0 +1,1074 @@
+    let other_group = ctx.get_new_group();
+
+    let mode = Mode::from_bits_truncate(0o555) | Mode::S_ISUID | Mode::S_ISGID;
+    let mode_without_setbits = mode & !Mode::S_ISUID & !Mode::S_ISGID;
+
+    assert!(lchown(&file, Some(user.uid), Some(group.gid)).is_ok());
+
+    lchmod(&file, mode).unwrap();
+
+    let file_stat = symlink_metadata(&file).unwrap();
+    assert_eq!(file_stat.mode() as mode_t & ALLPERMS_SETBITS, mode.bits());
+    assert_eq!(file_stat.uid() as uid_t, user.uid.as_raw());
+    assert_eq!(file_stat.gid() as gid_t, group.gid.as_raw());
+
+    ctx.as_user(&user, Some(&[group.gid, other_group.gid]), || {
+        assert!(lchown(&file, Some(user.uid), Some(other_group.gid)).is_ok());
+    });
+
+    let file_stat = symlink_metadata(&file).unwrap();
+    assert_eq!(
+        file_stat.mode() as mode_t & ALLPERMS_SETBITS,
+        mode_without_setbits.bits()
+    );
+    assert_eq!(file_stat.uid() as uid_t, user.uid.as_raw());
+    assert_eq!(file_stat.gid() as gid_t, other_group.gid.as_raw());
+
+    lchmod(&file, mode).unwrap();
+
+    let file_stat = symlink_metadata(&file).unwrap();
+    assert_eq!(file_stat.mode() as mode_t & ALLPERMS_SETBITS, mode.bits());
+    assert_eq!(file_stat.uid() as uid_t, user.uid.as_raw());
+    assert_eq!(file_stat.gid() as gid_t, other_group.gid.as_raw());
+
+    ctx.as_user(&user, Some(&[group.gid, other_group.gid]), || {
+        assert!(lchown(&file, None, Some(group.gid)).is_ok());
+    });
+    let file_stat = symlink_metadata(&file).unwrap();
+    assert_eq!(
+        file_stat.mode() as mode_t & ALLPERMS_SETBITS,
+        mode_without_setbits.bits()
+    );
+    assert_eq!(file_stat.uid() as uid_t, user.uid.as_raw());
+    assert_eq!(file_stat.gid() as gid_t, group.gid.as_raw());
+
+    lchmod(&file, mode).unwrap();
+
+    ctx.as_user(&user, Some(&[group.gid, other_group.gid]), || {
+        assert!(lchown(&file, None, None).is_ok());
+    });
+    let file_stat = symlink_metadata(&file).unwrap();
+    let actual_mode = file_stat.mode() as mode_t & ALLPERMS_SETBITS;
+    assert!(actual_mode == mode.bits() || actual_mode == mode_without_setbits.bits());
+    assert_eq!(file_stat.uid() as uid_t, user.uid.as_raw());
+    assert_eq!(file_stat.gid() as gid_t, group.gid.as_raw());
+}
+
+crate::test_case! {
+    /// successful chown(2) call (except uid and gid equal to -1) updates ctime.
+    update_ctime_success, serialized, root => [Regular, Dir, Fifo, Block, Char, Socket]
+}
+fn update_ctime_success(ctx: &mut SerializedTestContext, ft: FileType) {
+    let file = ctx.create(ft).unwrap();
+    let (user, group) = ctx.get_new_entry();
+    let other_group = ctx.get_new_group();
+
+    assert_ctime_changed(ctx, &file, || {
+        assert!(chown(&file, Some(user.uid), Some(group.gid)).is_ok());
+    });
+    let file_stat = metadata(&file).unwrap();
+    assert_eq!(file_stat.uid() as uid_t, user.uid.as_raw());
+    assert_eq!(file_stat.gid() as gid_t, group.gid.as_raw());
```

Comment:
Other tests already establish that `chown` changes the user.  We don't need to reverify that here.  In fact, it's harmful to do so, because if a file system regresses in a way that causes `chown` to fail to update the user, then the pjdfstest operator will see a failure for `update_ctime_success` and think that there's a failure related to ctime.

I suggest removing these assertions.  There are probably other places in the file where you can remove them too.
```suggestion
```